zsh-system-clipboard.zsh

```diff
@@ -40,11 +40,11 @@ if [[ "$ZSH_SYSTEM_CLIPBOARD_METHOD" == "" ]]; then
 			fi
 			;;
 		linux*|freebsd*)
-			if [[ "$DISPLAY" == "" ]]; then
+			if _zsh_system_clipboard_command_exists wl-copy && [[ "$WAYLAND_DISPLAY" != "" ]]; then
 				ZSH_SYSTEM_CLIPBOARD_METHOD="wlc"
-			elif _zsh_system_clipboard_command_exists xclip; then
+			elif _zsh_system_clipboard_command_exists xclip && [[ "$DISPLAY" != "" ]]; then
 				ZSH_SYSTEM_CLIPBOARD_METHOD="xcc"
-			elif _zsh_system_clipboard_command_exists xsel; then
+			elif _zsh_system_clipboard_command_exists xsel && [[ "$DISPLAY" != "" ]]; then
 				ZSH_SYSTEM_CLIPBOARD_METHOD="xsc"
 			else
 				_zsh_system_clipboard_suggest_to_install 'wl-clipboard / xclip / xsel'
```

Comment:
I tested this on my system on a tty without GUI, with `wl-copy` installed and I got this `suggest_to_install` message, as expected. This proves to me once more that it's hard to be accurate with these environment variables checks. The behavior I expect in such an environment as I tested is that running the `zsh-system-clipboard` will print me "`missing $DISPLAY`" errors or alike into the line editor - straight from the `{wl-copy,x{sel,clip}}` commands.

Reply:
We could print that on `[[ "$SSH_TTY" == "" ]] && [[ "$XDG_SESSION_TYPE" == tty ]]`, at this point if it fails to correctly detect environment at least it won't choose the wrong tool.